src/graphql/node/control.rs

```diff
@@ -99,37 +99,60 @@ impl NodeControlMutation {
         }
 
         if let Some(ref target_agents) = apply_scope.agents {
-            let agent_manager = ctx.data::<BoxedAgentManager>()?;
-            if let Err(e) = notify_agents(
-                agent_manager,
-                &target_agents.updates,
-                &target_agents.disables,
-            )
-            .await
-            {
-                error!("Failed to notify agents for node {i} to be updated. However, the failure does not affect the node application operation.\nDetails:\n{e:?}",);
-            }
+            let hostname = {
+                let store = crate::graphql::get_store(ctx).await?;
+                let node_map = store.node_map();
+                let (node, _) = node_map.get_by_id(i)?.ok_or_else(|| {
+                    async_graphql::Error::new(format!("Node with ID {i} not found"))
+                })?;
+                node.profile.map(|p| p.hostname).unwrap_or_default()
+            };
 
-            info!(
-                "[{}] Node ID {i} - Node's agents are notified to be updated.\n{:?}",
-                chrono::Utc::now(),
-                node.agents.iter().filter_map(|agent| {
-                    if target_agents.updates.contains(&agent.key.as_str()) {
-                        Some(format!(
-                            "\nAgent key: {}, Config: {:?}, Draft: {:?}",
-                            agent.key, agent.config, agent.draft
-                        ))
-                    } else {
-                        None
-                    }
-                })
-            );
+            if hostname.is_empty() {
+                info!(
+                    "Node ID {i} - Node's agents are not notified because the hostname is empty."
+                );
+            } else {
+                let agent_manager = ctx.data::<BoxedAgentManager>()?;
+                if let Err(e) = notify_agents(
+                    agent_manager,
+                    hostname.as_str(),
+                    &target_agents.updates,
+                    &target_agents.disables,
+                )
+                .await
+                {
+                    error!("Failed to notify agents for node {i} to be updated. However, the failure does not affect the node application operation.\nDetails:\n{e:?}",);
+                }
+
+                let agent_update_details = format_agent_updates(&node, target_agents);
+                info!(
+                    "[{}] Node ID {i} - Node's agents are notified to be updated.{agent_update_details}",
+                    chrono::Utc::now(),
+                );
+            }
         }
 
         Ok(ApplyNodeResponse { id, giganto_draft })
     }
 }
 
+fn format_agent_updates(node: &Node, target_agents: &NotificationTarget) -> String {
```

Comment:
Could you explain why the log message includes all the content of the agents' `config`s and `draft`s? Some of these items are usually not visible to users, as the UI only shows what can be modified by them, but they become visible because of this message. If you're considering keeping a history of all the changes, I think it should be stored in the database rather than in log messages.

Reply:
This log was created to follow this guide

> [Central server] logs the apply actions along with settings config, settings_draft draft, and the timestamp.

Reply:
It looks like I wrote the sentence. I apologize for the confusion caused by using the term "log" too loosely. When I used "log," I was thinking of its literal meaning—saving information—rather than in the syslog style. I intended that the specific method of storing could be decided based on the context. In this case, I don’t think the `config`s and `draft`s should be written to the log. As for how to handle them, I believe we can decide that later.

Reply:
Logs on `config` and `draft` are removed.

Reply:
> Logs on `config` and `draft` are removed.

I still see them.

Reply:
I am sorry for the mistake. 🙇 It's done from my side now.